### PR TITLE
Add the ability to diff

### PR DESCRIPTION
**Pull request by rsheeter**
Handy when hacking on things

### CODE REVIEW

File: src/picosvg/geometric_types.py [hunk 1]
```diff
@@ -44,13 +44,13 @@ def __sub__(self, other: _PointOrVec) -> _PointOrVec:
...
             return self._sub_vec(other)
-        return NotImplemented
+        raise ValueError(f"Other must be a Point or Vector, not a {type(other)}")
 
     def __add__(self, other: "Vector") -> "Point":
         """Return Point translated by other Vector"""
         if isinstance(other, Vector):
             return self.__class__(self.x + other.x, self.y + other.y)
-        return NotImplemented
+        raise ValueError(f"Other must be a Vector, not a {type(other)}")
```
Comment (rsheeter) [inline src/picosvg/geometric_types.py:53]: Changes here are due to typechecker complaining; I don't want the return type to be Point or NotImplemented

Comment (anthrotype) [reply src/picosvg/geometric_types.py:53]: but the canonical way to implement math operator overrides is exactly to return NotImplemented when the other type isn't among the ones which the current method is capable of handling, this enables the interpreter to check if the same operator is implemented by the other side of the expression, see https://docs.python.org/3/library/numbers.html#implementing-the-arithmetic-operations

I'm surprised that pytype doesn't automatically support this..

**Comment by anthrotype** [comment]
"Add the ability to diff" -- before reading the code, I thought it was about diffing two svg files..

**Review by anthrotype — APPROVED** [review]
LGTM.

by the way, what are you hacking on?

File: src/picosvg/geometric_types.py (hunk 1 above)
Comment (anthrotype) [reply src/picosvg/geometric_types.py:53]: there's even an explicit test for this in pytype test suite:

https://github.com/google/pytype/blob/b6b6448dc562a7200326c92e75efeed203984e16/pytype/tests/test_cmp2.py#L53-L78

Comment (rsheeter) [reply src/picosvg/geometric_types.py:53]: Huh. TIL. I will revert this part.

**Comment by rsheeter** [comment]
I wanted to colorize Ewert. Horrible hackery @ https://github.com/rsheeter/colorize. Haven't slapped gradients on it yet but I still quite like the result:

image

**Pull request merged**